Source/core/Proxy.h

```diff
@@ -989,321 +1016,320 @@ namespace Core {
         return (l_Received);
     }
 
-    template <typename PROXYPOOLELEMENT>
-    class ProxyPoolType {
-    private:
-        template <typename ELEMENT>
-        using PoolElement = typename std::conditional<std::is_base_of<IReferenceCounted, ELEMENT>::value != 0, ProxyService<ELEMENT>, ProxyObject<ELEMENT>>::type;
-
-        template <typename ELEMENT>
-        class ProxyObjectType : public PoolElement<ELEMENT> {
-        private:
-            template <typename... Args>
-            ProxyObjectType(ProxyPoolType<ELEMENT>* queue, Args&&... args)
-                : PoolElement<ELEMENT>(args...)
-                , _queue(*queue)
-            {
-                ASSERT(queue != nullptr);
-            }
-
-        public:
-            ProxyObjectType() = delete;
-            ProxyObjectType(const ProxyObjectType<ELEMENT>&) = delete;
-            ProxyObjectType<ELEMENT>& operator=(const ProxyObjectType<ELEMENT>&) = delete;
+    struct IProxyContainerElement : public virtual IReferenceCounted {
+        virtual ~IProxyContainerElement() {}
```

Comment:
out of curiousity why not ~IProxyContainerElement() override = default as we seem to do nowadays?

Reply:
No reason, it just did not  attention and should be updated. Will create a fix for this.